enterprise-suite/scripts/install-es.sh

```diff
@@ -151,13 +161,49 @@ else
     should_upgrade=false
 fi
 
-#echo "CREDS pre-use: $( ls -l $CREDS )"
-if [ "true" == "$should_upgrade" ]; then
-    debug helm upgrade "$ES_HELM_NAME" "$chart_ref" \
-        "$HELM_CREDENTIALS_ARG" \
-        $@
+if [ "true" == "$should_export" ]; then
+    # We need to grab any --version setting and pull it out of the command line args
+    VERSION_ARG=""
+    REST=()
+    if [ "true" = "$has_version" ] ; then
+        while [[ $# -gt 0 ]]
+        do
+            arg=$1
+            case $arg in
+                # Seems helm will accept "--version=foo" as well as the documented "--version foo"
+                --version=*)
+                    VERSION_ARG="$arg"
+                    shift
+                    ;;
+                --version)
+                    VERSION_ARG="--version $2"
+                    shift
+                    shift
+                    ;;
+                *)
+                    REST+=("$arg")
+                    shift
+                    ;;
+            esac
+        done
+        set -- "${REST[@]}" # restore the other positional parameters
+    fi
+
+    TDIR=$( mktemp -d 2>/dev/null || mktemp -d -t 'install-es-tdir' )
+    debug helm fetch -d $TDIR "$VERSION_ARG" "$chart_ref"
+    debug helm template --name "$ES_HELM_NAME" --namespace "$ES_NAMESPACE" \
+        "$@" \
+        $TDIR/${ES_CHART}*.tgz > "$ES_EXPORT_FILE"
```

Comment:
It's getting quite gnarly so I think we should stick this all into a function.

Reply:
Shred the gnar dude!

I tried moving that stuff to a function but it got even hairier cuz I was modifying $@ (which changed inside the function of course).  Leaving as is for now.